### DISCUSSION TITLE
Are training instances stored or purged post training?

### DISCUSSION THREAD

**Discussion by royr2** [online-ml/river#599] [Q&A] [answered]
Hi! 
Came across this library and played around with it a little bit - love how easy it is to get started with it. I am trying to figure out if any training instances are stored for future use somewhere within the model object? For my specific use case, I cannot store any data once it has been used to incrementally train the model (for legal reasons). 

Thanks!

**Comment by smastelini** [comment, marked as answer]
Hi @royr2. Most of River's learning models only keep data statistics. There are, however, cases where a window of recent examples is kept. This includes, for instance, k Nearest Neighbors Classifier/Regressor.

**Reply by royr2** [reply]
Appreciate the response @smastelini!
